recipes/hdf5/all/conanfile.py

```diff
@@ -171,11 +170,8 @@ def generate(self):
         tc.variables["HDF5_BUILD_CPP_LIB"] = self.options.enable_cxx
```

Comment:
Hi @SpaceIm, this comment is for line 164 above

```python
tc.variables["HDF5_INSTALL_INCLUDE_DIR"] = os.path.join(self.package_folder, "include", "hdf5").replace("\\", "/")
```

I'm running into the error mentioned in conan-io/conan#13402 while trying to build the hdf5 recipe at current tip of conan-center-index using conan 2.0.1; `conan create ...` works but `conan install ...` fails because `self.package_folder` is `None` within `generate()`.

Does the above line need to be modified for v2?